sstables/sstables.cc

```diff
@@ -1939,6 +1939,32 @@ bool sstable::has_component(component_type f) const {
     return _recognized_components.contains(f);
 }
 
+std::optional<bool> sstable::originated_on_this_node() const {
+    if (_version < version_types::me) {
```

Comment:
just quickly going through this patch, tomorrow I'll give it the attention it deserves. But I think we can change originated_on_this_node() to just return bool, where true = definitely originated and false = don't know. I see users default to false anyway when return value is not engaged.

Reply:
This is the direction I implemented it initially, but the function threw an exception in cases it wasn't sure.  That didn't work since it might be used early on, upon resharding of the system keyspace, and before the local host_id is loaded.
I think that returning false in cased we're not sure is confusing.  A function shouldn't assume the logic of its callers, as it might be used in the future in a different way.

Reply:
but it's fine as long as it's documented.

Reply:
the function wouldn't be assuming the logic of its callers, it's just that no caller so far needs to distinguish between "don't know" or "false". And if we document it, it's fine. Nobody will take "don't know" (which can be either false or unknown) as true.

Reply:
but this is a minor issue, was just a recommendation to make interface simpler.

Reply:
but on the other hand, optional boolean allows false to not be overloaded with another meaning.